flycheck-ledger.el

```diff
@@ -64,5 +65,72 @@ otherwise don't be pedantic."
 
 (add-to-list 'flycheck-checkers 'ledger)
 
+(defun flycheck-ledger-zero-last-position-of-account (account buffer)
+  "Return (LINE . COL) of last occurence of ACCOUNT in BUFFER.
+
+Return nil if ACCOUNT can't be found in BUFFER."
+  (with-current-buffer buffer
+    (save-restriction
+      (save-excursion
+        (goto-char (point-max))
+        (when (search-backward account nil t)
+          (cons (line-number-at-pos (point))
+                (1+ (- (point) (line-beginning-position)))))))))
+
+(defun flycheck-ledger-zero-error-parser (output checker buffer)
+  "Return errors found in OUTPUT.
+
+CHECKER is a `flycheck-ledger-zero' checker.
+
+_BUFFER is ignored.
+
+Return a list of parsed errors and warnings (as `flycheck-error'
+objects)."
+  (let ((errors (list))
+        (buffer (current-buffer)))
+    (save-match-data
+      (with-temp-buffer
+        (insert output)
+        (goto-char (point-min))
+        (while (re-search-forward "^ *\\(.*\\)  \\(.*\\)$" nil t)
+          (let* ((amount (match-string-no-properties 1))
+                 (account (match-string-no-properties 2))
+                 (message (format "Account %s should have zero value but has %s"
+                                  account amount))
+                 (position (flycheck-ledger-zero-last-position-of-account account buffer))
+                 (line (or (car position) 1))
+                 (column (or (cdr position) 0)))
+            (push
+             (flycheck-error-new
+              :line line :column column :message message
+              :checker checker :level 'error
+              :filename (buffer-file-name buffer) :buffer buffer)
+             errors)))))
+    errors))
+
+(flycheck-def-option-var flycheck-ledger-zero-accounts nil ledger-zero
+  "Whether to check account names, tags, and payees from cleared transactions."
+  :type '(repeat string))
+
+(flycheck-define-command-checker 'ledger-zero
+  "A checker for ledger  files, throwing errors when some accounts are non-zero."
+  :command `("ledger"
+             "-f" source-inplace
+             "balance"
+             ,@flycheck-ledger-zero-accounts)
+  :error-parser #'flycheck-ledger-zero-error-parser
+  :verify
+  (lambda (checker)
+    (let ((has-accounts (> (length flycheck-ledger-zero-accounts) 0)))
+      (list
+       (flycheck-verification-result-new
+        :label "has zero accounts"
+        :message (if has-accounts "yes" "no")
```

Comment:
Minor nit.  "has zero accounts" refers to the `-zero-accounts` variable, I get it.  But it sounds strange when reading, as if it checked for the number of Ledger accounts in my file.

Another way of putting it would be:
```emacs-lisp
:label "accounts"
:message (if has-accounts flycheck-ledger-zero-accounts "none")
```

Reply:
done